### PR TITLE
Guess operator

### PR DESCRIPTION
**Pull request by strobelm**
Hi, I just finished the guess operator :-). It follows closely the implementation of Cinderella, so it should be bulletproof. 

I backed most of the code with tests.

### CODE REVIEW

File: src/js/libcs/Operators.js [hunk 1]
```diff
@@ -5097,3 +5097,256 @@ evaluator.removeelement$1 = function(args, modifs) {
     if (arg.ctype === "geo") removeElement(arg.value.name);
     else console.log("argument of removeelement is undefined or not of type <geo>");
 };
+
+
+evaluator.guess$1 = function(args, modifs) {
+    function pslq(x) {
+        const c = Math.PI * Math.E;
```
Comment (montaga) [inline src/js/libcs/Operators.js]: I do not want to comment this line (Did the algorithm detect `pi*e` as rational? ;-))

File: src/js/libcs/Operators.js [hunk 2]
```diff
@@ -5097,3 +5097,256 @@ evaluator.removeelement$1 = function(args, modifs) {
...
+    function extractsquare(a) {
+        for (let n = 3000; n > 0; n--)
+            if (a % (n * n) === 0)
+                return n;
+        return 1;
+    }
+
+
+    /**
+     * @return a string representation of a complex number
```
Comment (montaga) [inline src/js/libcs/Operators.js]: There must be an error somewhere here because `guess(i*pi)` gives `NaN/NaN-i*1/NaN*sqrt(NaN)`

**Comment by kortenkamp** [comment]
> And why does `guess(exp(pi)-pi)` not give 20? ;-)

Because it isn't 20? I am quite happy it doesn't, but maybe I am misinterpreting the ;-)?

**Comment by montaga** [comment]
> > And why does `guess(exp(pi)-pi)` not give 20? ;-)
> 
> Because it isn't 20? I am quite happy it doesn't, but maybe I am misinterpreting the ;-)?

Sure it is not. I was referring to https://xkcd.com/217/

**Comment by kortenkamp** [comment]
> Sure it is not. I was referring to https://xkcd.com/217/

Didn't knew that one. thanks!

**Comment by montaga** [comment]
With the last commit the bug occuring while guessing complex numbers should be fixed.

**Pull request merged**